platform/android/variant/callable_jni.cpp

```diff
@@ -0,0 +1,117 @@
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "callable_jni.h"
+
+#include "jni_utils.h"
+
+#include "core/error/error_macros.h"
+#include "core/object/object.h"
+
+static Callable _generate_callable(JNIEnv *p_env, jlong p_object_id, jstring p_method_name, jobjectArray p_parameters) {
+	Object *obj = ObjectDB::get_instance(ObjectID(p_object_id));
+	ERR_FAIL_NULL_V(obj, Callable());
+
+	String str_method = jstring_to_string(p_method_name, p_env);
+
+	int count = p_env->GetArrayLength(p_parameters);
+
+	Variant *args = (Variant *)alloca(sizeof(Variant) * count);
+	const Variant **argptrs = (const Variant **)alloca(sizeof(Variant *) * count);
+
+	for (int i = 0; i < count; i++) {
+		jobject jobj = p_env->GetObjectArrayElement(p_parameters, i);
+		ERR_FAIL_NULL_V(jobj, Callable());
+		memnew_placement(&args[i], Variant(_jobject_to_variant(p_env, jobj)));
+		argptrs[i] = &args[i];
+		p_env->DeleteLocalRef(jobj);
+	}
+
+	return Callable(obj, str_method).bindp(argptrs, count);
```

Comment:
I think this will lead to a memory leak, but I'm not 100% sure.

The reason I suspect it is that I don't think the destructors for each `Variant` in `args` will run automatically. This is fine for any variant type where all the data is part of the `Variant` class (because this uses `alloca()` to put their data on the stack), but some types refer to memory elsewhere (for example: `String`s, `RefCounted` object pointers or any of the `Packed*Array` types) and use reference counting, which won't be decremented when this function exits, leaving the ref count artificially inflated by one.

So, I think we need to add some code at the end that manually calls `~Variant()`, like:

```suggestion
	Variant *args = (Variant *)alloca(sizeof(Variant) * count);
	const Variant **argptrs = (const Variant **)alloca(sizeof(Variant *) * count);

	for (int i = 0; i < count; i++) {
		jobject jobj = p_env->GetObjectArrayElement(p_parameters, i);
		ERR_FAIL_NULL_V(jobj, Callable());
		memnew_placement(&args[i], Variant(_jobject_to_variant(p_env, jobj)));
		argptrs[i] = &args[i];
		p_env->DeleteLocalRef(jobj);
	}

	Callable ret = Callable(obj, str_method).bindp(argptrs, count);
	
	for (int i = 0; i < count; i++) {
		args[i].~Variant();
	}
	
	return ret;
```

I don't think this should cause any problems, because `.bindp()` will copy each `Variant`, which will increment the ref count (if there is a refcount for that particular variant type).